client/cli/src/node_key.rs

```diff
@@ -0,0 +1,225 @@
+// Copyright 2017-2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+use sc_network::{
+	self,
+	config::{
+		NodeKeyConfig,
+	},
+};
+use sp_core::H256;
+use names::{Generator, Name};
+use regex::Regex;
+use std::{path::{Path, PathBuf}, str::FromStr};
+use crate::error;
+use crate::params::{NodeKeyParams, NodeKeyType};
+
+/// The maximum number of characters for a node name.
+const NODE_NAME_MAX_LENGTH: usize = 32;
+
+/// The file name of the node's Ed25519 secret key inside the chain-specific
+/// network config directory, if neither `--node-key` nor `--node-key-file`
+/// is specified in combination with `--node-key-type=ed25519`.
+const NODE_KEY_ED25519_FILE: &str = "secret_ed25519";
+
+pub fn generate_node_name() -> String {
```

Comment:
What does this function do in this module? It seems out of place.

Reply:
True! Will move it back! I tried to split a bit because `lib.rs` was too dense